EIPS/eip-3540.md

```diff
@@ -79,58 +81,95 @@ The container starts with the EOF header:
+
+container  := header, body
+header := magic, version, kind_type, type_size, kind_code, num_code_sections, code_size+, kind_data, data_size, terminator
+body := type_section, code_section+, data_section
+type_section := (inputs, outputs, max_stack_height)+
+```
+
+*note: `,` is a concatenation operator and `+` should be interpreted as "one or more" of the preceding item*
+
+##### Header
 
-The section kinds for EOF version 1 are defined as follows. The list may be extended in future versions.
+| name              | length   | value         | description                                                                        |
+|-------------------|----------|---------------|------------------------------------------------------------------------------------|
+| magic             | 2 bytes  | 0xEF00        | EOF prefix                                                                         |
+| version           | 1 byte   | 0x01          | EOF version                                                                        |
+| kind_type         | 1 byte   | 0x01          | kind marker for EIP-4750 type section header                                       |
+| type_size         | 2 bytes  | 0x0003-0xFFFF | uint16 denoting the length of the type section content                             |
+| kind_code         | 1 byte   | 0x02          | kind marker for code size section                                                  |
+| num_code_sections | 2 bytes  | 0x0001-0xFFFF | uint16 denoting the number of the code sections                                    |
+| code_size         | 2 bytes  | 0x0001-0xFFFF | uint16 denoting the length of the code section content                             |
+| kind_data         | 1 byte   | 0x03          | kind marker for data size section                                                  |
+| data_size         | 2 bytes  | 0x0001-0xFFFF | uint16 integer denoting the length of the data section content                     |
+| terminator        | 1 byte   | 0x00          | marks the end of the header                                                        |
 
-| section_kind | meaning                                        |
-|--------------|------------------------------------------------|
-| 0            | *reserved for section headers terminator byte* |
-| 1            | code                                           |
-| 2            | data                                           |
+##### Body
+
+| name             | length   | value         | description |
+|------------------|----------|---------------|-------------|
+| type_section     | variable | n/a       | stores EIP-4750 and EIP-5450 code section metadata  |
+| inputs           | 1 byte   | 0x00-0x7F | number of stack elements the code section consumes  |
+| outputs          | 1 byte   | 0x00-0x7F | number of stack elements the code section returns   |
+| max_stack_height | 1 bytes  | 0x00-0xFF | `ceil(max_stack_height / 8)` during execution       |
```

Comment:
There are still concerns if the value should be stored in the EOF at all. I think for now we should store uncompressed value (i.e. 2 bytes).

Reply:
What are the concerns of storing this in EOF? I think not storing it in the type section is basically a non-starter as it makes it much more complicated for bootstrapping nodes to get the type information.

I will update to use the uncompressed value though. We should consider increasing the the max code size limit at some point to make up for this extra data.